src/figuresClasses.ts

```diff
@@ -1,19 +1,97 @@
 export interface Figure {
-
+  color: 'red' | 'green' | 'blue';
+  shape: 'triangle' | 'circle' | 'rectangle';
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  color: Figure['color'];
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: Figure['color'], a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be greater than 0');
+    }
 
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Invalid side lengths for a triangle');
+    }
+
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
```

Comment:
try to use [typescript shorthand](https://dev.to/satansdeer/typescript-constructor-shorthand-3ibd)